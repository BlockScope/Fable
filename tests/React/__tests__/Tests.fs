module Fable.Tests.React

open Fable.Jester
open Fable.ReactTestingLibrary

Jest.describe("React tests", (fun () ->

    Jest.test("ReactComponent renders correctly", (fun () ->
        let elem = RTL.render(App.Counter())
        Jest.expect(elem.container).toMatchSnapshot()
    ))

    Jest.test("ReactComponent state works", (fun () ->
        let elem = RTL.render(App.Counter())
        let header = elem.getByTestId "header"
        let button = elem.getByTestId "button-increment"
        Jest.expect(header).toHaveTextContent("0")
        RTL.fireEvent.click(button)
        Jest.expect(header).toHaveTextContent("1")
    ))
))