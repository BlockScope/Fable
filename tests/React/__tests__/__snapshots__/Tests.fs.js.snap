// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React tests Counter renders correctly 1`] = `
<div>
  <div
    style="padding: 10px;"
  >
    <h1
      data-testid="header"
    >
      0
    </h1>
    <button
      data-testid="button-increment"
    >
      Increment
    </button>
  </div>
</div>
`;

exports[`React tests SpreadSheet renders correctly 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <td />
        <th>
          A
        </th>
        <th>
          B
        </th>
        <th>
          C
        </th>
        <th>
          D
        </th>
        <th>
          E
        </th>
        <th>
          F
        </th>
        <th>
          G
        </th>
        <th>
          H
        </th>
        <th>
          I
        </th>
        <th>
          J
        </th>
        <th>
          K
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          1
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          2
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          3
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          4
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          5
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          6
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          7
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          8
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          9
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          10
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          11
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          12
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          13
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          14
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
      <tr>
        <th>
          15
        </th>
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
        <td
          style="background: white;"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
